pyximages/numpy_fancy_indexing.py

```diff
@@ -0,0 +1,105 @@
+'''source code for figure
+   numpy_fancy_indexing.(png/pdf) used in Sect. 1.3.1.7 Fancy indexing
+
+   Requirement: PyX>=0.14  (pip installable)
+
+'''
+   
+import sys, os
+from math import cos, radians, sin
+from pyx import canvas, color, path, text, unit
+
+def markbox(x, y, boxcolor, linewidthfactor=0.1, boxsize=1):
+    '''mark box by a colored line
+
+       the drawing is done on the global canvas called c
+    '''
+    linewidth = linewidthfactor*boxsize
+    p = (path.rect(x, y, boxsize, boxsize)
+         + path.rect(x+linewidth, y+linewidth,
+                     boxsize-2*linewidth, boxsize-2*linewidth).reversed()
+         )
+    c.fill(p, [boxcolor])
+
+ncols = 6
+nrows = ncols
+boxsize = 1
+angle = radians(40)
+reducedboxsize = 0.65*boxsize
+
+text.set(text.LatexRunner)
+text.preamble(r'''\usepackage[T1]{fontenc}
+                  \usepackage{bera}
+                  \renewcommand*\familydefault{\ttdefault}
+                  \usepackage{color}
+                  \definecolor{ex1}{rgb}{0, 0, 0.7}
+                  \definecolor{ex2}{rgb}{0, 0.6, 0}
+                  \definecolor{ex3}{rgb}{0.7, 0, 0}''')
+unit.set(xscale=1.2)
+
+c= canvas.canvas()
```

Comment:
Nitpick: this is not PEP8 (missing whitespace before assignment).

Also, it might help readability if the variable "c" was called "canvas"

Reply:
Forget about the canvas thing.